### PR TITLE
Add support for ComplexDataVector to Pypp tests

### PR DESCRIPTION
**Pull request by moxcodes**
## Proposed changes

Python (and numpy) routines gracefully handle complex values, so it's just a matter of bringing the values from the `ComplexDataVector` over to the numpy structures.

### Types of changes:

- [ ] New feature

### Component:

- [ ] Code

### Code review checklist

- [ ] The PR passes all checks, including unit tests, `clang-tidy` and `IWYU`.
  For instructions on how to perform the CI checks locally refer to the [Dev
  guide on the Travis CI](https://spectre-code.org/travis_guide.html).
- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).

### CODE REVIEW

**Review by nilsdeppe — DISMISSED** [review]

**Review by osheamonn — DISMISSED** [review]

**Comment by moxcodes** [comment]
squashed in iwyu and clang-tidy changes to make Travis happy.

File: tests/Unit/Pypp/CheckWithRandomValues.hpp [hunk 1]
```diff
@@ -143,7 +143,7 @@ void check_with_random_values_impl(
     std::index_sequence<ResultIs...> /*index_return_types*/,
     std::index_sequence<ArgumentIs...> /*index_argument_types*/,
     std::index_sequence<MemberArgsIs...> /*index_member_args*/,
-    TagsList /*meta*/, const double epsilon = 1.0e-12) {
+    TagsList /*meta*/, double epsilon = 1.0e-12) {
```
Comment (nilsdeppe) [inline tests/Unit/Pypp/CheckWithRandomValues.hpp]: this is fixed more correctly in #1389 so my vote is let's ignore it :) clang-tidy is wrong here

Comment (moxcodes) [reply tests/Unit/Pypp/CheckWithRandomValues.hpp]: Alright, sounds good. Removed.

**Review by nilsdeppe — APPROVED** [review]

**Review by osheamonn — APPROVED** [review]

**Comment by nilsdeppe** [comment]
clang-tidy failure is false positive address in #1389

**Pull request merged**